dmwmclient/datasvc.py

```diff
@@ -494,3 +494,172 @@ async def missingfiles(self, human_readable=None, **params):
+        version          phedex version
+        update_since     updated since this time
+        detail           'y' or 'n', default 'n'. show "code" information at file level *
+        """
+        resjson = await self.jsonmethod("agents", **params)
+        out = []
+        if human_readable is not None and type(human_readable) is not bool:
+            print("Wrong human_readable parameter type")
+            df = pandas.json_normalize(out)
+            return df
+        elif human_readable is None or human_readable is False:
+            for _node in resjson["phedex"]["node"]:
+                for _agent in _node["agent"]:
+                    out.append(
+                        {
+                            "Node": _node["node"],
+                            "Host": _node["host"],
+                            "Agent_name": _node["name"],
+                            "Agent_label": _agent["label"],
+                            "Time_update": _agent["time_update"],
+                            "state_dir": _agent["state_dir"],
+                            "version": _agent["version"],
+                        }
+                    )
+            df = pandas.json_normalize(out)
+            return format_dates(df, ["Time_update"])
+        elif human_readable is True:
+            for _node in resjson["phedex"]["node"]:
+                for _agent in _node["agent"]:
+                    out.append(
+                        {
+                            "Node": _node["node"],
+                            "Host": _node["host"],
+                            "Agent name": _node["name"],
+                            "Agent label": _agent["label"],
+                            "Time update": _agent["time_update"],
+                            "Directory": _agent["state_dir"],
+                            "Version": _agent["version"],
+                        }
+                    )
+            df = pandas.json_normalize(out)
+            return format_dates(df, ["Time update"])
+
+    async def BlockLatency(self, human_readable=None, **params):
+        """Show authentication state and abilities
+        Parameters
+        ----------
+        ability        authorization ability.  If passed then the nodes (from TMDB)
+                       that the user is allowed to use "ability" for are returned.
+        require_cert   if passed then the call will die if the user is not
+                       authenticated by certificate
+        require_passwd if passed then the call will die if the user is not
+                       authenticated by password
+        """
+        resjson = await self.jsonmethod("blocklatency", **params)
+        out = []
+        if human_readable is not None and type(human_readable) is not bool:
+            print("Wrong human_readable parameter type")
+            df = pandas.json_normalize(out)
+            return df
+        elif human_readable is None or human_readable is False:
```

Comment:
In all of these, wouldn't the code be more compact if the `human_readable` switch was inside the nested loop rather than at the top level?